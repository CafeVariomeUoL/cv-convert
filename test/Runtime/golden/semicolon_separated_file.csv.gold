[
{
    "age": 69,
    "diagnosis_age": 1,
    "diagnosis": 0,
    "first_symptoms_age": 2,
    "subject_id": 0,
    "sex": "F"
} , {
    "age": 15,
    "diagnosis_age": 2,
    "diagnosis": 0,
    "first_symptoms_age": 0,
    "subject_id": 1,
    "sex": "M"
} , {
    "age": 4,
    "diagnosis_age": 4,
    "diagnosis": 0,
    "first_symptoms_age": 0,
    "subject_id": 2,
    "sex": "M"
} , {
    "age": 12,
    "diagnosis_age": 2,
    "diagnosis": 0,
    "first_symptoms_age": 4,
    "subject_id": 3,
    "sex": "F"
} , {
    "age": 8,
    "diagnosis_age": 14,
    "diagnosis": 0,
    "first_symptoms_age": 0,
    "subject_id": 4,
    "sex": "F"
} , {
    "age": 31,
    "diagnosis_age": 0,
    "diagnosis": 0,
    "first_symptoms_age": 0,
    "subject_id": 5,
    "sex": "M"
} , {
    "age": 32,
    "diagnosis_age": 7,
    "diagnosis": 0,
    "first_symptoms_age": 1,
    "subject_id": 6,
    "sex": "F"
} , {
    "age": 16,
    "diagnosis_age": 5,
    "diagnosis": 0,
    "first_symptoms_age": 12,
    "subject_id": 7,
    "sex": "F"
} , {
    "age": 25,
    "diagnosis_age": 3,
    "diagnosis": 0,
    "first_symptoms_age": 0,
    "subject_id": 8,
    "sex": "F"
} , {
    "age": 18,
    "diagnosis_age": 3,
    "diagnosis": 0,
    "first_symptoms_age": 3,
    "subject_id": 9,
    "sex": "M"
}
]