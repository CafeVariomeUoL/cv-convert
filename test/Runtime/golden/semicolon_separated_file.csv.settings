{
  "processFunction": "const ret = {\n  subject_id: row.i,\n  sex: row.pat_sex,\n  age: Lib.calculate_age(new Date(row.pat_birth_date)),\n  diagnosis_age: Math.round(row.pat_diagnosis_age/365),\n  first_symptoms_age: Math.round(row.pat_onset_age/365),\n  diagnosis: 0\n}\n\nif(row.pat_diagnosis_age_sel == \"99\") delete ret.diagnosis_age\nif(row.pat_onset_age_sel == \"99\") delete ret.first_symptoms_age\nreturn ret;",
  "libraryFunctions": "Lib.calculate_age = function(dob) { \n    var diff_ms = Date.parse('2020-09-01') - dob.getTime();\n    var age_dt = new Date(diff_ms); \n  \n    return Math.abs(age_dt.getUTCFullYear() - 1970);\n}\n",
  "tsInterface": "export default interface Row {\n    subject_id:          any;\n    sex:                 \"M\"|\"F\"|\"U\";\n    age:                 number;\n    diagnosis_age?:      number;\n    first_symptoms_age?: number;\n    diagnosis:           number|number[];\n}",
  "jsonSchema": {
    "$schema": "http://json-schema.org/draft-07/schema#",
    "definitions": {
      "Row": {
        "$ref": "#/definitions/default"
      },
      "default": {
        "title": "default",
        "type": "object",
        "properties": {
          "subject_id": {
            "title": "subject_id"
          },
          "sex": {
            "enum": [
              "F",
              "M",
              "U"
            ],
            "type": "string",
            "title": "sex"
          },
          "age": {
            "type": "number",
            "title": "age"
          },
          "diagnosis_age": {
            "type": "number",
            "title": "diagnosis_age"
          },
          "first_symptoms_age": {
            "type": "number",
            "title": "first_symptoms_age"
          },
          "diagnosis": {
            "anyOf": [
              {
                "type": "array",
                "items": {
                  "type": "number"
                }
              },
              {
                "type": "number"
              }
            ],
            "title": "diagnosis"
          }
        },
        "additionalProperties": false,
        "required": [
          "age",
          "diagnosis",
          "sex",
          "subject_id"
        ]
      }
    },
    "$ref": "#/definitions/default"
  }
}