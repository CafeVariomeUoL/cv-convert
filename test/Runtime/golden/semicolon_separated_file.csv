pat_birth_date;pat_sex;pat_onset_age;pat_diagnosis_age
1951-07;F;700;366
2005-01;M;41;700
2016-07;M;5;1400
2008-08;F;1601;881
2012-06;F;123;5010
1989-09;M;17;32
1987-10;F;441;2415
2004-04;F;4400;1648
1995-04;F;0;1011
2001-11;M;1055;1045
